extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -0,0 +1,325 @@
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link WeatherUndergroundHandler} is responsible for handling the
+ * weather things created to use the Weather Underground Service.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+public class WeatherUndergroundHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(WeatherUndergroundHandler.class);
+
+    private static final String URL = "http://api.wunderground.com/api/%APIKEY%/%FEATURES%/%SETTINGS%/q/%QUERY%.json";
+    private static final String FEATURE_CONDITIONS = "conditions";
+    private static final String FEATURE_FORECAST10DAY = "forecast10day";
+    private static final String FEATURE_GEOLOOKUP = "geolookup";
+    private static final Set<String> USUAL_FEATURES = Sets.newHashSet(FEATURE_CONDITIONS, FEATURE_FORECAST10DAY);
+
+    private static final int DEFAULT_REFRESH_PERIOD = 30;
+
+    private WeatherUndergroundJsonData weatherData;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private Gson gson;
+
+    public WeatherUndergroundHandler(Thing thing) {
+        super(thing);
+
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing WeatherUnderground handler.");
+
+        WeatherUndergroundConfiguration config = getConfigAs(WeatherUndergroundConfiguration.class);
+        logger.debug("config apikey = {}", config.apikey);
+        logger.debug("config location = {}", config.location);
+        logger.debug("config language = {}", config.language);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+        String errors = "";
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errors += " Parameter 'apikey' must be configured.";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.location) == null) {
+            errors += " Parameter 'location' must be configured.";
+            validConfig = false;
+        }
+        if (config.language != null) {
+            String lang = StringUtils.trimToEmpty(config.language);
+            if (lang.length() != 2) {
+                errors += " Parameter 'language' must be 2 letters.";
+                validConfig = false;
+            }
+        }
+        if (config.refresh != null && config.refresh < 5) {
+            errors += " Parameter 'refresh' must be at least 5 minutes.";
+            validConfig = false;
+        }
+        errors = errors.trim();
+
+        if (validConfig) {
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
You are mixing up two things here.

We indeed had some issues with the startup when we first introduced the asynchronous parsing of the XMLs. And if there still are bugs in the framework, we should raise them and fix them there, not in the bindings.

There also were a couple of buggy bindings, not setting a status at all under some circumstances. Then these bindings should be fixed. 

The [contract](https://www.eclipse.org/smarthome/documentation/concepts/things.html) is pretty clear: INITIALIZING may take as long as it needs, UNKNOWN means initialization is complete and the handler is ready to work. The framework is relying on that!